scapy/arch/windows/__init__.py

```diff
@@ -72,11 +72,39 @@ def _encapsulate_admin(cmd):
     # rights, which will execute the command
     return "Start-Process PowerShell -windowstyle hidden -Wait -Verb RunAs -ArgumentList '-command &{%s}'" % cmd
 
+def _hack_windows_subprocess():
+    """HACK: python 2.7 file descriptors.
+
+    This magic hack fixes https://bugs.python.org/issue19575
+    by adding HANDLE_FLAG_INHERIT to all already opened file descriptors.
+    """
+    # See https://github.com/secdev/scapy/issues/1136
+    import stat
+    from ctypes import windll, wintypes
+    from msvcrt import get_osfhandle
+
+    HANDLE_FLAG_INHERIT = 0x00000001
+
+    for fd in range(100):
```

Comment:
Could you pass it as argument with default value?
It fixes my problem, but I can't anticipate other issues, so it'd be nice to have an API in place.

Reply:
Updated